lightningd/peer_control.c

```diff
@@ -248,11 +253,83 @@ static bool get_peer_gossipfd_reply(struct subd *subd, const u8 *msg,
 	return true;
 }
 
-static void get_gossip_fd_for_reconnect(struct lightningd *ld,
-					const struct pubkey *id,
-					u64 unique_id,
-					int peer_fd,
-					const struct crypto_state *cs)
+static void get_gossip_fd_for_channeld_reconnect(struct lightningd *ld,
+						 const struct pubkey *id,
+						 u64 unique_id,
+						 int peer_fd,
+						 const struct crypto_state *cs)
+{
+	struct getting_gossip_fd *ggf = tal(ld, struct getting_gossip_fd);
+	u8 *req;
+
+	ggf->peer_fd = peer_fd;
+	ggf->id = *id;
+	ggf->cs = *cs;
+
+	/* FIXME: set sync to `initial_routing_sync` */
+	req = towire_gossipctl_get_peer_gossipfd(ggf, unique_id, true);
+	subd_req(ggf, ld->gossip, take(req), -1, 1,
+		 get_peer_gossipfd_channeld_reply, ggf);
+}
+
+static bool get_peer_gossipfd_closingd_reply(struct subd *subd, const u8 *msg,
```

Comment:
Duplicates a lot of code from `get_peer_gossipfd_channeld_reply` except which daemon is started afterwards, can we join the two?